File: Second Year/PDOO/Napakalaki/src/NapakalakiGame/Treasure.java
/*  
 *  Programación y diseño orientado a objetos
 *  Grado en Ingeniería Informática
 * 
 *  
 *
 *  Angela Castro Jiménez
 *  Francisco Carrillo Pérez
 */
package NapakalakiGame;

public class Treasure implements Card{
    private String name;
    private int goldCoins;
    private int minBonus;
    private int maxBonus;
    private TreasureKind type;
    
    public Treasure (String n, int g,int min, int max, TreasureKind t){
        name = n;
        goldCoins = g;
        minBonus = min;
        maxBonus = max;
        type= t;
    }
    
    public String getName(){
        return name;
    }
    
    public int getGoldCoins(){
        return goldCoins;
    }
    
    public int getMinBonus(){
        return minBonus;
    }
    
    public int getMaxBonus(){
        return maxBonus;
    }
    
    public TreasureKind getType(){
        return type;
    }
    
     public String toString(){
        return "Nombre: "+name
                +".\nMonedas: " + Integer.toString(goldCoins)
                +".\nBonus Minimo: " +minBonus
                +"\nBonus Maximo"+ maxBonus
                +"\nTipo de Tesoro: "+ type;
        }
     
     @Override
     public int getBasicValue(){
         return this.getMinBonus();
     }
    
     @Override
     public int getSpecialValue(){
         return this.getMaxBonus();
     }
}

File: Second Year/PDOO/Napakalakirb/lib/Treasure.rb
# encoding: UTF-8
#
#   Programación y diseño orientado a objetos
#   Grado en Ingeniería Informática
#
#  Ángela Castro Jimenez
#  Francisco Carrillo Pérez

module Napakalaki
  
require_relative'Card'
class Treasure < Card
  attr_accessor :name,:goldCoins,:minBonus,:maxBonus,:type
  def initialize(name,goldCoins,minBonus,maxBonus,type)
        @name = name
        @goldCoins = goldCoins
        @minBonus = minBonus
        @maxBonus = maxBonus
        @type = type
        
  end
  
  def self.Treasure(n,g,min,max,t)
    new(n,g,min,max,t)
  end
  
  def getName()
     @name
  end
  def getGoldCoins()
    @goldCoins
  end
  
 def getMinBonus()
   @minBonus
 end
 
 def getMaxBonus()
   @maxBonus
 end
 
 def getType()
     @type
 end
 def to_s
      "#{@name}. GoldCoins: #{@goldCoins}. Minbonus #{@minBonus}. MaxBonus #{@maxBonus}. Type #{@type}"
    end
end

 def basic_value
        return getMinBonus;
    end

    def special_value
      return getMaxBonus;
    end
end
